### DISCUSSION TITLE
How can I extract the displayed top features from a beeswarm plot?

### DISCUSSION THREAD

**Discussion by manavka** [shap/shap#3053] [Q&A] [answered]
Is there a way for me to extract the top features displayed on the beeswarm plot? Specifically, I want to extract the selected features on the y axis. I have tried to access the matplotlib plot used on the backend but have had no success in extracting the selected features. I have attached a photo for reference, I want to extract the 9 features on the y axis. Any help is appreciated!
![Beeswarm](https://github.com/slundberg/shap/assets/71353846/87cb9bfc-c766-4cb4-a4f2-961fbaffb109)

**Comment by thatlittleboy** [comment, marked as answer]
@manavka 

The most straightforward way to do it is to reach into the figure and pull out the y labels, as you've rightly pointed out. (The beeswarm plot _can_ involve hierarchical clustering, so it's not quite the same as just sorting the shap values and taking the top x.)

```python
shap.plots.beeswarm(shap_values, show=False)
top_features = [
    t.get_text() for t in reversed(plt.gca().yaxis.get_majorticklabels())
]
print(f"Top features: {top_features}")
plt.show()
```

![image](https://github.com/slundberg/shap/assets/30731072/6a4a29eb-9c2c-4894-a8e3-f2d24cdc0b29)

ps: you need to reverse the ytick labels because the origin of the plot is at the lower left corner. (The top feature is furthest from the origin).
